homeassistant/components/evohome/climate.py

```diff
@@ -119,10 +134,39 @@ def __init__(self, evo_broker, evo_device) -> None:
 
         self._preset_modes = None
 
-    async def _set_tcs_mode(self, op_mode: str) -> None:
+    async def async_tcs_svc_request(self, service: str, data) -> None:
+        """Process a service request (system mode) for a controller."""
+        if service == SVC_SET_SYSTEM_MODE:
+            mode = data[ATTR_SYSTEM_MODE]
+        else:  # otherwise it is SVC_RESET_SYSTEM
+            mode = EVO_RESET
+
+        allowed_system_modes = self._evo_device.allowedSystemModes
+        try:
+            attrs = [m for m in allowed_system_modes if m["systemMode"] == mode][0]
+        except IndexError:
+            raise KeyError(f"'{mode}' mode is not supported by this system")
```

Comment:
If you want to use a built-in error, use `ValueError` as it's an incorrect value.

Reply:
OK, and there's another place I will fix too.